Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1539 @@
+
+    H5::Attribute attribute = parentGroup.openAttribute(NX_CLASS);
+    std::string attributeValue;
+    attribute.read(attribute.getDataType(), attributeValue);
+
+    return attributeValue == attrVal;
+  }
+
+  bool hasAttributeInDataSet(
+      const std::string dataSetName, const std::string &attrName,
+      const std::string &attrVal,
+      const fullH5Path &pathToGroup /*where the dataset lives*/) {
+
+    H5::Attribute attribute;
+    H5::Group parentGroup = openfullH5Path(pathToGroup);
+    H5::DataSet dataSet = parentGroup.openDataSet(dataSetName);
+    attribute = dataSet.openAttribute(attrName);
+    std::string attributeValue;
+    attribute.read(attribute.getDataType(), attributeValue);
+
+    return attributeValue == attrVal;
+  }
+
+  bool hasNXAttributeInDataSet(const std::string dataSetName,
+                               const std::string &attrVal,
+                               const fullH5Path &pathToGroup) {
+    H5::Attribute attribute;
+    H5::Group parentGroup = openfullH5Path(pathToGroup);
+    H5::DataSet dataSet = parentGroup.openDataSet(dataSetName);
+    attribute = dataSet.openAttribute(NX_CLASS);
+    std::string attributeValue;
+    attribute.read(attribute.getDataType(), attributeValue);
+
+    return attributeValue == attrVal;
+  }
+
+private:
+  H5::H5File m_file;
+
+}; // HDF5FileTestUtility
+
+// RAII: Gives a clean file destination and removes the file when
+// handle is out of scope. Must be stack allocated.
+class ScopedFileHandle {
+
+public:
+  ScopedFileHandle(const std::string &fileName) {
+
+    const auto temp_dir = boost::filesystem::temp_directory_path();
+    auto temp_full_path = temp_dir;
+    temp_full_path /= fileName;
+
+    // Check proposed location and throw std::invalid argument if file does
+    // not exist. otherwise set m_full_path to location.
+
+    if (boost::filesystem::is_directory(temp_dir)) {
+      m_full_path = temp_full_path;
+
+    } else {
+      throw std::invalid_argument("failed to load temp directory: " +
+                                  temp_dir.generic_string());
+    }
+  }
+
+  std::string fullPath() const { return m_full_path.generic_string(); }
+
+  ~ScopedFileHandle() {
+
+    // file is removed at end of file handle's lifetime
+    if (boost::filesystem::is_regular_file(m_full_path)) {
+      boost::filesystem::remove(m_full_path);
+    }
+  }
+
+private:
+  boost::filesystem::path m_full_path; // full path to file
+
+  // prevent heap allocation for ScopedFileHandle
+protected:
+  static void *operator new(std::size_t); // prevent heap allocation of scalar.
+  static void *operator new[](std::size_t); // prevent heap allocation of array.
+};
+
+} // namespace
+
+//---------------------------------------------------------------------
+
+class NexusGeometrySaveTest : public CxxTest::TestSuite {
+
+private:
+  std::pair<std::unique_ptr<Mantid::Geometry::ComponentInfo>,
+            std::unique_ptr<Mantid::Geometry::DetectorInfo>>
+      m_instrument;
```

Comment:
Do we need this global instrument? In lots of test cases you don't use it. At least rename it to be more specific about what it is `m_testSimpleInstrumentWithRotation`

Reply:
removed, since its only called three times